pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -163,11 +163,22 @@ func (cc *cadvisorClient) MachineInfo() (*cadvisorapi.MachineInfo, error) {
 	return cc.GetMachineInfo()
 }
 
-func (cc *cadvisorClient) DockerImagesFsInfo() (cadvisorapiv2.FsInfo, error) {
-	return cc.getFsInfo(cadvisorfs.LabelDockerImages)
+func (cc *cadvisorClient) ImagesFsInfo(runtime string) (cadvisorapiv2.FsInfo, error) {
+	var label string
+
+	switch runtime {
+	case "docker":
+		label = cadvisorfs.LabelDockerImages
+	case "rkt":
+		label = cadvisorfs.LabelRktImages
+	default:
+		return cadvisorapiv2.FsInfo{}, fmt.Errorf("ImagesFsInfo: unknown runtime: %v", runtime)
+	}
+
+	return cc.getFsInfo(label)
 }
 
-func (cc *cadvisorClient) RootFsInfo() (cadvisorapiv2.FsInfo, error) {
+func (cc *cadvisorClient) RootFsInfo(_ string) (cadvisorapiv2.FsInfo, error) {
```

Comment:
I'm not particurally happy with it, but its needed as my previous approach of having a single function with a cadvisorfs label passed in (which would include root) doesn't work for unsupported archs (i.e. darwin, windows....)